test/proposal_test.rb

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env rspec
+# encoding: utf-8
+
+# Copyright (c) [2016] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative "spec_helper"
+require_relative "test_refinements"
+require "storage"
+require "storage/proposal"
+require "storage/fake_probing"
+require "storage/fake_device_factory"
+
+describe Yast::Storage::Proposal do
+  describe "#propose" do
+    using Yast::Storage::TestRefinements
+
+    def input_file_for(name)
+      File.join(DATA_PATH, "input", "#{name}.yml")
+    end
+
+    def output_file_for(name)
+      File.join(DATA_PATH, "output", "#{name}.yml")
+    end
+
+    def fake_scenario(scenario)
+      fake_probing = Yast::Storage::FakeProbing.new
+      devicegraph = fake_probing.devicegraph
+      Yast::Storage::FakeDeviceFactory.load_yaml_file(devicegraph, input_file_for(scenario))
+      fake_probing.to_probed
```

Comment:
"fake" is not a well-known device graph. It exists only when created as FakeProbing.

"probed" does have a well-defined meaning, as the name implies. At some point we do have to trigger hardware probing in all normal cases (except in a testing environment), so it makes perfect sense to have something like this with a predefined meaning. Otherwise that probing stuff would have to be scattered all over the place.